test/e2e/performanceprofile/functests/2_performance_update/tuned_deferred.go

```diff
@@ -0,0 +1,355 @@
+					Expect(err).To(BeNil())
+					Expect(out).To(BeTrue())
+					rebootNode(poolName)
+					verifyImmediateApplication(profile, tuned, workerCNFNodes, tc.KernelSHMMNI)
+				}
+			},
+			Entry("[test_id:78115] Verify deferred Always with first-time profile change", TunedProfileConfig{
+				DeferMode:         "always",
+				ProfileChangeType: "first-time",
+				ExpectedBehavior:  "deferred",
+				KernelSHMMNI:      "8192",
+			}),
+			Entry("[test_id:78116] Verify deferred Always with in-place profile update", TunedProfileConfig{
+				DeferMode:         "always",
+				ProfileChangeType: "in-place",
+				ExpectedBehavior:  "deferred",
+				KernelSHMMNI:      "4096",
+			}),
+			Entry("[test_id:78117] Verify deferred Update with first-time profile change", TunedProfileConfig{
+				DeferMode:         "update",
+				ProfileChangeType: "first-time",
+				ExpectedBehavior:  "immediate",
+				KernelSHMMNI:      "8192",
+			}),
+			Entry("[test_id:78118] Verify deferred Update with in-place profile update", TunedProfileConfig{
+				DeferMode:         "update",
+				ProfileChangeType: "in-place",
+				ExpectedBehavior:  "deferred",
+				KernelSHMMNI:      "4096",
+			}),
+			Entry("[test_id:78119] Verify deferred with No annotation with first-time profile change", TunedProfileConfig{
+				DeferMode:         "",
+				ProfileChangeType: "first-time",
+				ExpectedBehavior:  "immediate",
+				KernelSHMMNI:      "8192",
+			}),
+			Entry("[test_id:78120] Verify deferred Never mode with in-place profile update", TunedProfileConfig{
+				DeferMode:         "never",
+				ProfileChangeType: "in-place",
+				ExpectedBehavior:  "immediate",
+				KernelSHMMNI:      "4096",
+			}),
+		)
+	})
+
+	AfterAll(func() {
+		tuned = getTunedProfile(name)
+		if tuned != nil {
+			Expect(testclient.ControlPlaneClient.Delete(context.TODO(), tuned)).To(Succeed())
+		}
+		profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Unable to fetch latest performance profile err: %v", err))
+		profile.Spec.NUMA = &performancev2.NUMA{
+			TopologyPolicy: initialPolicy,
+		}
+
+		By("Updating the performance profile")
+		profiles.UpdateWithRetry(profile)
+
+		By(fmt.Sprintf("Applying changes in performance profile and waiting until %s will start updating", poolName))
+		profilesupdate.WaitForTuningUpdating(context.TODO(), profile)
+
+		By(fmt.Sprintf("Waiting when %s finishes updates", poolName))
+		profilesupdate.WaitForTuningUpdated(context.TODO(), profile)
+	})
+
+})
+
+func stringPtr(s string) *string {
+	return &s
+}
+
+func getTunedProfile(tunedName string) *tunedv1.Tuned {
+	GinkgoHelper()
+
+	tunedList := &tunedv1.TunedList{}
+	var matchedTuned *tunedv1.Tuned
+
+	Eventually(func(g Gomega) {
+		g.Expect(testclient.DataPlaneClient.List(context.TODO(), tunedList, &client.ListOptions{
+			Namespace: components.NamespaceNodeTuningOperator,
+		})).To(Succeed())
+
+		g.Expect(len(tunedList.Items)).To(BeNumerically(">", 1))
```

Comment:
`g.Expect(tunedList.Items).ToNot(BeEmpty())`

Reply:
Resolved with this Gomega assertion, Thanks